domain/modeldefaults/service/service.go

```diff
@@ -183,24 +226,66 @@ func ProviderDefaults(
 // in a model.
 func (s *Service) providerDefaults(
 	ctx context.Context,
-	uuid coremodel.UUID,
-) (modeldefaults.Defaults, error) {
-	modelCloudType, err := s.st.ModelCloudType(ctx, uuid)
+	cloudUUID cloud.UUID,
+) (modeldefaults.Defaults, schema.Checker, error) {
+	modelCloudType, err := s.st.CloudType(ctx, cloudUUID)
 	if err != nil {
-		return nil, errors.Errorf(
-			"getting model %q cloud type to extract provider model config defaults: %w",
-			uuid, err,
+		return nil, nil, errors.Errorf(
+			"getting %q cloud type to extract provider model config defaults: %w",
+			cloudUUID, err,
 		)
 	}
 
-	defaults, err := ProviderDefaults(ctx, modelCloudType, s.modelConfigProviderGetter)
+	defaults, checker, err := ProviderDefaults(ctx, modelCloudType, s.modelConfigProviderGetter)
 	if err != nil {
-		return nil, errors.Errorf(
-			"getting model %q provider defaults: %w", uuid, err,
+		return nil, nil, errors.Errorf(
+			"getting cloud %q provider defaults: %w", cloudUUID, err,
 		)
 	}
 
-	return defaults, nil
+	return defaults, checker, nil
+}
+
+// CloudDefaults returns the default attribute details for a specified cloud.
+// It returns an error satisfying [clouderrors.NotFound] if the cloud doesn't exist.
+func (s *Service) CloudDefaults(ctx context.Context, cloudName string) (modeldefaults.ModelDefaultAttributes, error) {
+	cloudUUID, err := s.st.GetCloudUUID(ctx, cloudName)
+	if err != nil {
+		return nil, errors.Errorf("getting cloud UUID for cloud %q: %w", cloudName, err)
+	}
+	return s.cloudDefaults(ctx, cloudUUID)
+}
+
+// UpdateModelConfigDefaultValues saves the specified default attribute details for a cloud or region.
+// It returns an error satisfying [clouderrors.NotFound] if the cloud doesn't exist.
+func (s *Service) UpdateModelConfigDefaultValues(ctx context.Context, updateAttrs map[string]interface{}, cloudRegion modeldefaults.CloudRegion) error {
```

Comment:
Is there a different error if the region does not exist?

Reply:
No. We don't do anything with this error. We would just be masking it with a generic not found anyway.